From d5787710821e62bcdbbe1d3bf00aa2c2dc6a5032 Mon Sep 17 00:00:00 2001
From: Rajendra Nayak <rnayak@ti.com>
Date: Mon, 7 Oct 2013 14:00:21 +0530
Subject: [PATCH 0557/1115] ARM: OMAP4+: Fix PRM device instance offsets for
 OMAP5 and DRA7

The PRM device instance offsets for OMAP5 and DRA7 are different
from that on OMAP4. While this was fixed in the
omap4_prminst_global_warm_sw_reset() function, it was fixed only
partially for the register read part. Fix the offsets in the rest
of the read/wriites done in the function.

Reported-by: Yan Liu <yan-liu@ti.com>
Signed-off-by: Rajendra Nayak <rnayak@ti.com>
(cherry picked from commit ff020498cf2ad8013ad826ffe9e7d2ace119c445)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 arch/arm/mach-omap2/prminst44xx.c |    6 ++----
 1 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-omap2/prminst44xx.c b/arch/arm/mach-omap2/prminst44xx.c
index 6334b96..5fb9bfa 100644
--- a/arch/arm/mach-omap2/prminst44xx.c
+++ b/arch/arm/mach-omap2/prminst44xx.c
@@ -182,12 +182,10 @@ void omap4_prminst_global_warm_sw_reset(void)
 	v = omap4_prminst_read_inst_reg(OMAP4430_PRM_PARTITION, dev_inst,
 					OMAP4_PRM_RSTCTRL_OFFSET);
 	v |= OMAP4430_RST_GLOBAL_WARM_SW_MASK;
-	omap4_prminst_write_inst_reg(v, OMAP4430_PRM_PARTITION,
-				 OMAP4430_PRM_DEVICE_INST,
+	omap4_prminst_write_inst_reg(v, OMAP4430_PRM_PARTITION, dev_inst,
 				 OMAP4_PRM_RSTCTRL_OFFSET);
 
 	/* OCP barrier */
-	v = omap4_prminst_read_inst_reg(OMAP4430_PRM_PARTITION,
-				    OMAP4430_PRM_DEVICE_INST,
+	v = omap4_prminst_read_inst_reg(OMAP4430_PRM_PARTITION, dev_inst,
 				    OMAP4_PRM_RSTCTRL_OFFSET);
 }
-- 
1.7.5.4

